Raven.NET.Core.Tests/WatcherTests/RavenTypeWatcherTests.cs

```diff
@@ -0,0 +1,226 @@
+using Raven.NET.Core.Observers;
+using Raven.NET.Core.Observers.Interfaces;
+using Raven.NET.Core.Providers.Interfaces;
+using Raven.NET.Core.Static;
+using Raven.NET.Core.Subjects;
+using Raven.NET.Core.Tests.SubjectTests;
+using Shouldly;
+using Xunit;
+
+namespace Raven.NET.Core.Tests.WatcherTests;
+
+[Collection("Tests")]
+public class RavenTypeWatcherTests
+{
+    private Fixture _fixture;
+    private readonly IRavenTypeWatcher sut;
+    private readonly Mock<IRavenProvider> _ravenProvider;
+    private readonly Mock<IRavenSettingsProvider> _ravenSettingsProvider;
+    
+    public RavenTypeWatcherTests()
+    {
+        _fixture = new Fixture();
+        _ravenProvider = new Mock<IRavenProvider>();
+        _ravenSettingsProvider = new Mock<IRavenSettingsProvider>();
+        sut = new RavenTypeWatcher(_ravenProvider.Object, _ravenSettingsProvider.Object);
+    }
+    
+    [Fact]
+    void Create_Should_AddWatcherToCacheWithDefaultConfiguration_When_NoConfigFileSpecified()
+    {
+        //Arrange
+        var watcherName = "TestWatcher";
+        _ravenProvider.Setup(x => x.AddRaven(It.IsAny<string>(), It.IsAny<IRavenTypeWatcher>(), typeof(TestSubjectEntity))).Callback(() =>
+        {
+            RavenCache.RavenTypeWatcherCache.TryAdd(typeof(TestSubjectEntity), sut);
+        });
+        
+        //Act
+        sut.Create<TestSubjectEntity>(watcherName, nameof(TestSubjectEntity.Value),subject => true);
+        
+        //Assert
+        RavenCache.RavenTypeWatcherCache.ShouldContainKey(typeof(TestSubjectEntity));
+        var settings = (sut as RavenTypeWatcher)?._ravenSettings;
+        settings.LogLevel.ShouldBe(LogLevel.Warning);
+        settings.AutoDestroy.ShouldBe(false);
+        settings.BackgroundWorker.ShouldBe(true);
+        _ravenProvider.Verify(x => x.AddRaven(watcherName, sut, typeof(TestSubjectEntity)));
+        _ravenSettingsProvider.Verify(x => x.GetRaven(watcherName));
+    }
+    
+    [Fact]
+    void Create_Should_AddWatcherToCacheWithFileConfiguration_When_ConfigFileSpecified()
+    {
+        //Arrange
+        var watcherName = "TestWatcher";
+        var settingsMock = _fixture.Create<RavenSettings>();
+        settingsMock.AutoDestroy = true;
+        settingsMock.LogLevel = LogLevel.Trace;
+        
+        _ravenProvider.Setup(x => x.AddRaven(It.IsAny<string>(), It.IsAny<IRavenTypeWatcher>(), typeof(TestSubjectEntity))).Callback(() =>
+        {
+            RavenCache.RavenTypeWatcherCache.TryAdd(typeof(TestSubjectEntity), sut);
+        });
+
+        _ravenSettingsProvider.Setup(x => x.GetRaven(watcherName)).Returns(settingsMock);
+        
+        //Act
+        sut.Create<TestSubjectEntity>(watcherName, nameof(TestSubjectEntity.Value),subject => true);
+        
+        //Assert
+        RavenCache.RavenTypeWatcherCache.ShouldContainKey(typeof(TestSubjectEntity));
+        var settings = (sut as RavenTypeWatcher)?._ravenSettings;
+        settings.LogLevel.ShouldBe(LogLevel.Trace);
```

Comment:
Github is hinting error, nice feature but here maybie this is true

Reply:
This is mocked object, so its kind of impossible for this to be null, but I`ve added additional assertion (just to be 1212% sure :D ), this should remove github hint